### DISCUSSION TITLE
Splitting large Markdown note

### DISCUSSION THREAD

**Discussion by dm5n** [wheelercj/note-splitter#20] [Q&A]
Hi. I'd appreciate any advice about this workflow:

I was previously using Joplin app to collect an archive of quotes extracted from academic literature. Then I found the Obsidian md app and it's non-linear knowledge graph offering a visual way to organise and retrieve information.

I have a single large Joplin note that I've exported to Markdown format in order to import into Obsidian md.

The note is in the form of a numbered list of hundreds of items (each item consisting of academic quotes, images, references and paths to local files).

I would like to split the note item into individual .md files. The numbering is not required.

Could I use Note-splitter to automate this?

Thankyou

**Comment by wheelercj** [comment]
Hello, and thanks for your interest! Although note-splitter is still in development, I believe it will be easy to split notes by numbered list items or any other category of markdown syntax. I'm not sure when these features will be complete, but it will probably take somewhere within 1-4 weeks.

**Comment by dm5n** [comment]
Excellent, will check in then. Thankyou.

**Reply by wheelercj** [reply]
Good morning, here's a progress update: I have a working prototype of the application! You can see [the source code here](https://github.com/wheelercj/note-splitter/tree/numeric-prototype).  

It has a few temporary limitations:  
1. This version only allows splitting by all ordered list items, and doesn't have a GUI.
2. There is not yet a way to run this without running the Python source code. By any chance, do you know how to run Python code? I will be looking into how to make the application run on devices that don't have Python at some point in the near future.
3. All new files' names are 14-digit numbers in the time format YYYYMMDDhhmmss.md, the first new file's name is the current time, and each new file's name increases the seconds by one.
4. All new files appear in the same folder as the source file.
5. The numbers at the start of each line of the list are also copied into the new files.

The source file is not changed in any way.

Here's a way to run Note Splitter, if you have Python and pip:
1. make sure pyyaml is installed with `pip install pyyaml`
2. in the application's root folder, use `python3 note-splitter/main.py`
3. follow the instructions that appear

There is still much work to do, including making a user interface with lots of settings to choose from. Although we aren't using version numbers yet, I would consider this an early beta. If you are able to try this version or if you have any questions/concerns about the description of it above, please let me know what you think! All feedback is important, especially at this stage of development.

**Reply by dm5n** [reply]
Hi, thanks for the update.

Having a few difficulties. I'm on macOS 10.14.6 and have installed Python 3 and pip.

Form the Terminal, I `cd` to the note-splitter-master directory and using the command: `python note-splitter/main.py`  gives error msg:
`python: can't open file 'note-splitter/main.py: [Errno 2] No such file or directory`

Using the command `python3 note-splitter/main.py`
gives:

```
frontmatter:
{'title': 'this is a file with frontmatter', 'description': 'Frontmatter is used to add structured data to a note. More details here https://assemble.io/docs/YAML-front-matter.html'}

global tags:
['#first-tag', '#second-tag']

<class 'tokens.Text'>:



<class 'tokens.Header'>:
# sample markdown


<class 'tokens.Text'>:
#first-tag #second-tag


<class 'tokens.Text'>:



<class 'tokens.UnorderedListItem'>:
* bullet point 1


<class 'tokens.UnorderedListItem'>:
* bullet point 2


<class 'tokens.Text'>:



<class 'tokens.Text'>:
here is text


<class 'tokens.Text'>:



<class 'tokens.OrderedListItem'>:
1. ordered


<class 'tokens.OrderedListItem'>:
2. list


<class 'tokens.Text'>:



<class 'tokens.Header'>:
## second header


<class 'tokens.Text'>:
#third-tag <- this tag should not be considered a global tag because it's below a header of a level > 1


<class 'tokens.Text'>:



<class 'tokens.CodeFence'>:
```python


<class 'tokens.Text'>:
print('this code is inside a code block')


<class 'tokens.Text'>:
while True:


<class 'tokens.Text'>:
    print(eval(input('>>> ')))


<class 'tokens.CodeFence'>:
```


<class 'tokens.Text'>:



<class 'tokens.Header'>:
### third header


<class 'tokens.ToDo'>:
- [ ] to do


<class 'tokens.Done'>:
- [x] done


<class 'tokens.Text'>:



<class 'tokens.Blockquote'>:
> this is


<class 'tokens.Blockquote'>:
> a blockquote


<class 'tokens.Text'>:



<class 'tokens.Header'>:
#### fourth header


<class 'tokens.CodeFence'>:
```cpp


<class 'tokens.Text'>:
cout << "here's another code block";


<class 'tokens.CodeFence'>:
```


<class 'tokens.Text'>:
```

**Reply by wheelercj** [reply]
Hello, and thanks for the feedback! This should be fairly easy to fix. It looks like you got a version of the program that was just for testing somehow. Are you able to download the version of the program from the numeric-prototype branch? Here's the same link: https://github.com/wheelercj/note-splitter/tree/numeric-prototype

I didn't know the 3 in `python3 note-splitter/main.py` is still sometimes required, thanks for letting me know! (I just edited my earlier instructions in case someone else uses them.)

**Reply by dm5n** [reply]
Ok, it's working. Thanks and well done.

The split is not working well on my .md file. 

I think it's because [Joplin](https://joplinapp.org/) has nested the lists in a confusing way.  

Would you be interested in checking out the file? I'd appreciate any advice on how to fix it so it splits better. 

If so, I'd rather email it than attach it.

**Reply by wheelercj** [reply]
Yeah I'll take a look! You can email me at christopher.wheeler.320 at my.csun.edu
I will have to see how the lists are nested, but there's a chance I'll only have to make a tiny change to get it working for your file's format. The final version of our program will have tons of options for different file formats.

**Comment by dm5n** [comment]
Thankyou! Sent.

**Comment by dm5n** [comment]
Hi,

Which branch (Master/Dev/Numeric) should I download from to test the latest script?

Thanks

**Reply by wheelercj** [reply]
Hello! Currently, the [numeric-prototype branch](https://github.com/wheelercj/note-splitter/tree/numeric-prototype) is the only branch that's set up for normal use (for splitting by numbered list items only) rather than testing. We're still adding more features and the newer code on the other branches isn't ready for use yet.

Another feature we're working on that I don't think I mentioned to you yet is creating an index file with links to all the new files each time the program runs.
